BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java

```diff
@@ -95,17 +127,139 @@ public void onBindViewHolder(@NonNull NotesViewHolder holder, int position) {
+    holder.itemView.findViewById(R.id.voice_note_layout).setVisibility(View.VISIBLE);
+    ImageButton playButton = holder.getPlayNoteButton();
+    playButtons.add(playButton);
+    playButton.setVisibility(View.VISIBLE);
+
+    ImageButton stopButton = holder.getStopNoteButton();
+    stopButtons.add(stopButton);
+    stopButton.setVisibility(View.VISIBLE);
+
+    MediaPlayer mediaPlayer = new MediaPlayer();
+    mediaPlayers.add(mediaPlayer);
+
+    RelativeLayout voiceNoteLayout = holder.getVoiceNoteLayout();
+    voiceNoteLayout.setVisibility(View.VISIBLE);
+
+    SeekBarCompat progressBar = holder.getProgressBar();
+    progressBars.add(progressBar);
+    progressBar.setEnabled(false);
+
+    TextView playedTime = holder.getPlayedTime();
+    TextView totalTime = holder.getTotalTime();
+
+    SeekBarListener seekBarListener =
+        new SeekBarListener(activity, mediaPlayer, progressBar, playedTime);
+    seekBarListeners.add(seekBarListener);
+
+    setTotalTimes(noteItem, totalTime);
+
+    progressBar.setOnSeekBarChangeListener(seekBarListener);
+
+    setupMediaPlayerListeners(mediaPlayer, progressBar, playButton, stopButton, noteItem,
+        seekBarListener);
+  }
+
+  private void setTotalTimes(NoteItem noteItem, TextView totalTime) {
+    MediaMetadataRetriever mmr = new MediaMetadataRetriever();
+    mmr.setDataSource(noteModel.getNoteFilePath(noteItem.getId()));
+    String durationStr = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+    int millis = Integer.parseInt(durationStr);
+    String time = new SimpleDateFormat("mm:ss", Locale.getDefault())
+        .format(new Date(millis));
+    totalTime.setText(time);
+  }
+
+  private void setupMediaPlayerListeners(MediaPlayer mediaPlayer, SeekBarCompat progressBar,
+                                         ImageButton playButton,
+                                         ImageButton stopButton,
+                                         NoteItem noteItem, SeekBarListener seekBarListener) {
+    mediaPlayer.setOnCompletionListener(mp -> {
+      mp.reset();
+      seekBarListener.reset();
+      stopButton.setClickable(false);
+      setSelection(playButton, false, R.drawable.icon_play);
+      paused = false;
+    });
+
+    playButton.setOnClickListener(v -> {
+      ImageButton button = (ImageButton) v;
+      if (v.isSelected()) {
+        if (paused) {
+          mediaPlayerPosition = mediaPlayer.getCurrentPosition();
+          mediaPlayer.seekTo(mediaPlayerPosition);
+          mediaPlayer.start();
+          button.setImageResource(R.drawable.icon_pause);
+        } else {
+          mediaPlayer.pause();
+          button.setImageResource(R.drawable.icon_play);
+        }
+        paused = !paused;
+      } else {
+        resetPlayers();
+        startAudio(mediaPlayer, button, noteItem, seekBarListener);
+        stopButton.setClickable(true);
+      }
+      progressBar.setEnabled(true);
+    });
```

Comment:
Some new lines here?

```suggestion
    playButton.setOnClickListener(v -> {
      ImageButton button = (ImageButton) v;
      
      if (v.isSelected()) {
        if (paused) {
          mediaPlayerPosition = mediaPlayer.getCurrentPosition();
          mediaPlayer.seekTo(mediaPlayerPosition);
          mediaPlayer.start();
          button.setImageResource(R.drawable.icon_pause);
          
        } else {
          mediaPlayer.pause();
          button.setImageResource(R.drawable.icon_play);
        }
        
        paused = !paused;
      } else {
        resetPlayers();
        startAudio(mediaPlayer, button, noteItem, seekBarListener);
        stopButton.setClickable(true);
      }
      
      progressBar.setEnabled(true);
    });
```